src/lib.rs

```diff
@@ -172,6 +172,10 @@ pub enum DeepgramError {
     /// An unexpected error occurred in the client
     #[error("an unepected error occurred in the deepgram client: {0}")]
     InternalClientError(anyhow::Error),
+
+    /// A Deepgram API server response was not in the expected format.
+    #[error("The Deepgram API server response was not in the expected format: {0}")]
+    UnexpectedServerResponse(String),
```

Comment:
Change this from String to `anyhow::Error`, which is more flexible, and allows us to provide better-structured errors.  (Then at error construction time, you can use `anyhow!("missing request request_id")` and `anyhow!("malformed request_id: {:?}", request_id)`)

Reply:
fe7e71d10f9d